crates/sui-source-validation-service/src/lib.rs

```diff
@@ -417,7 +433,8 @@ pub async fn watch_for_upgrades(
                 info!("Saw failed transaction when listening to upgrades.")
             }
             None => {
-                bail!("Fatal: WebSocket connection lost while listening for upgrades. Shutting down server.")
+                error!("Fatal: WebSocket connection lost while listening for upgrades. Shutting down server.");
+                std::process::exit(1)
```

Comment:
I'll make this more robust in follow ups with a catch-up routine and/or automatic restart following our "kick the server" practices. Currently the server appears to be running robustly enough to manage manually and this code path isn't triggered all that often.